internal/cli/cmd/apply/apply.go

```diff
@@ -19,8 +19,54 @@ import (
 	kargosvcapi "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
+type applyOptions struct {
+	*option.Option
+
+	Filenames []string
+}
+
+// newApplyOptions creates a new options struct.
+func newApplyOptions(opt *option.Option) *applyOptions {
```

Comment:
Nit: I'm a big fan of constructor-like functions if they're either exported for use outside the package _or_ are used internally, but are still performing non-trivial initialization. Since this one is neither of those scenarios and is also called only from one place, I'd just as soon kill it. _But it is not a hill I would die on._

Reply:
I actually agree, and this was mainly added to wrap my head around factoring things out at the beginning. Have removed them.